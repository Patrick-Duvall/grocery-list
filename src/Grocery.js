import React from 'react'

const Grocery = ({ name, quantity, purchased, starred, onPurchase, onStar, onDelete}) => {
  return ( 
    <article className={`Grocery ${starred && 'starred'}`}>
      <h3>{name}</h3>
      <p>quantity: {quantity}</p>
    </article>
  )
}

export default Grocery;