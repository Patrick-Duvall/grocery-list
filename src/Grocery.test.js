import React from 'react'
import {shallow, mount } from 'enzyme';
import Grocery from './Grocery';

describe('Grocery', () => {

  it('should match the snapshot with provided data', () => {
    const wrapper = shallow(
      <Grocery 
        name="Bananas" 
        quantity="7"
        purchased="false"
        starred="false"
        onStar={jest.fn()}
        onPurchase={jest.fn()}
        onDelete={jest.fn()}
      />
    );

    expect(wrapper).toMatchSnapshot();
  })

  it('should nave a classname of tarred if starred', () => {
    const wrapper = shallow(
      <Grocery name='Bananas' starred={true} />
    )

    expect(wrapper.is('.starred')).toEqual(true)
  })
})