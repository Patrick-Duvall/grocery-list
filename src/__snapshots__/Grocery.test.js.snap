// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grocery should match the snapshot with provided data 1`] = `ShallowWrapper {}`;
